python/pyarrow/_dataset.pyx

```diff
@@ -1013,27 +1013,38 @@ cdef class ParquetReadOptions(_Weakrefable):
     dictionary_columns : list of string, default None
         Names of columns which should be dictionary encoded as
         they are read.
+    enable_parallel_column_conversion : bool, default False
+        EXPERIMENTAL: Parallelize conversion across columns. This option is
+        ignored if a scan is already parallelized across input files to avoid
+        thread contention. This option will be removed after support is added
+        for simultaneous parallelization across files and columns.
```

Comment:
I was going to comment: is it needed to add this option in the *public* dataset interface? Since we can enable/disable it under the hood depending on whether we are reading a single file or not, and since we do not plan to keep this keyword long term. 

But, currently we implement this logic in parquet.py, which of course needs to be able to pass it through to the dataset API, so it needs to be a keyword this way.